### PR TITLE
[PassBuilder] Replace `bool LTOPreLink` with `ThinOrFullLTOPhase Phase`

### PR DESCRIPTION
**Pull request by shiltian**
This will allow more fine-grained control in the future.

### CODE REVIEW

**Comment by shiltian** [comment]
* **#114564** <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/114564?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @shiltian and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: llvm/lib/Passes/PassBuilderPipelines.cpp [hunk 1]
```diff
@@ -1635,6 +1632,8 @@ PassBuilder::buildPerModuleDefaultPipeline(OptimizationLevel Level,
   // Emit annotation remarks.
   addAnnotationRemarksPass(MPM);
 
+  bool LTOPreLink = Phase == ThinOrFullLTOPhase::FullLTOPreLink ||
+                    Phase == ThinOrFullLTOPhase::ThinLTOPreLink;
```
Comment (nikic) [inline llvm/lib/Passes/PassBuilderPipelines.cpp:1636]: There's an `isLTOPreLink()` helper for this.

File: llvm/lib/Passes/PassBuilderPipelines.cpp [hunk 2]
```diff
@@ -2220,6 +2220,8 @@ ModulePassManager PassBuilder::buildO0DefaultPipeline(OptimizationLevel Level,
 
   invokeOptimizerLastEPCallbacks(MPM, Level);
 
+  bool LTOPreLink = Phase == ThinOrFullLTOPhase::FullLTOPreLink ||
+                    Phase == ThinOrFullLTOPhase::ThinLTOPreLink;
```
Comment (nikic) [inline llvm/lib/Passes/PassBuilderPipelines.cpp:2224]: Same here.

**Review by nikic — APPROVED** [review]
LGTM

**Pull request merged**